mailbox/store/src/main/java/org/apache/james/mailbox/store/search/SimpleMessageSearchIndex.java

```diff
@@ -132,14 +192,16 @@ private Set<MailboxMessage> searchResults(MailboxSession session, Mailbox mailbo
             // only fetching this uid range
             UidRange[] ranges = uidCrit.getOperator().getRange();
             for (UidRange r : ranges) {
-                Iterator<MailboxMessage> it = mapper.findInMailbox(mailbox, MessageRange.range(r.getLowValue(), r.getHighValue()), FetchType.METADATA, UNLIMITED);
+                FetchType fetchType = maxFetchType(FetchType.METADATA, getFetchTypeForSorts(query.getSorts()));

```

Comment:
Is better if we move fetchType outside the `for loop`

Reply:
Right, totally makes sense. I will fix it.